### PR TITLE
Shield bounce feature for death aircraft

### PR DESCRIPTION
**Pull request by aeoncleanse**
Based on Exotic_Retard's work, trying a new approach that will _hopefully_ end up a little simpler and more robust.

### CODE REVIEW

File: projectiles/ShieldCollider/ShieldCollider_script.lua [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                self:SetPosition(pos, true)
+                self.Plane:SetPosition(pos, true) -- Make sure the plane is above ground if its not
+                
+                self:SetVelocity(0, 0, 0) -- The plane is attached to our projectile, so that stops too
+                
+                self.Plane:OnImpact('Terrain', true) -- Tell the plane we hit land
+            end
+        elseif targetType == 'Water' then
+            self:DetachAll('anchor')
+            self.Plane:OnImpact('Terrain', true) -- Tell the plane we hit water
```
Comment (Exotic-Retard) [inline projectiles/ShieldCollider/ShieldCollider_script.lua]: should probably be water and not terrain. not sure if it was me or you who got this one wrong xD

File: projectiles/ShieldCollider/ShieldCollider_script.lua [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        
+        --applying our bounce velocity
+        vx = -stoppingPower*wx*dotProduct + vx 
+        vy = -stoppingPower*wy*dotProduct + vy
+        vz = -stoppingPower*wz*dotProduct + vz
+        
+        --so sometimes absurd values pop up, probably due to rounding errors or something, so we prevent huge speeds here
+        vx = math.min(7,vx)
+        vy = math.min(4,vy) -- less for y so we dont get planes flying into space
+        vz = math.min(7,vz)
```
Comment (Exotic-Retard) [inline projectiles/ShieldCollider/ShieldCollider_script.lua]: i realise now that this was totally silly. these only prevent it in one direction. need to use math.clamp:
```
vx = math.clamp(-7, vx, 7)
vy = math.clamp(-4, vy, 4) -- less for y so we dont get planes flying into space
vz = math.clamp(-7, vz, 7)
```

File: projectiles/ShieldCollider/ShieldCollider_script.lua (hunk 1 above)
Comment (aeoncleanse) [inline projectiles/ShieldCollider/ShieldCollider_script.lua]: Me :p

File: projectiles/ShieldCollider/ShieldCollider_script.lua (hunk 2 above)
Comment (aeoncleanse) [inline projectiles/ShieldCollider/ShieldCollider_script.lua]: I did wonder xD

**Pull request merged**